tests/device/simulator_adapter_test.py

```diff
@@ -69,3 +75,35 @@ def test_if_simulator_adapter_raises_exception_empty_command(device) -> None:
     exception_msg = 'Run simulation command is empty, please verify if it was generated properly.'
     with pytest.raises(TwisterRunException, match=exception_msg):
         device.flash_and_run(timeout=0.1)
+
+
+@mock.patch('asyncio.run', side_effect=subprocess.SubprocessError(1, 'Exception message'))
+def test_if_simulator_adapter_raises_exception_when_subprocess_raised_exception(patched_run, device):
```

Comment:
I know that technically Python can rise `Exception` not `Error`, but maybe better name for this test could be `test_if_simulator_adapter_raises_exception_when_subprocess_raised_an_error` just to distinguish this test from next one (`test_if_simulator_adapter_raises_exception_when_subprocess_raised_an_exception`) which is very similar,

Reply:
Updated